### PR TITLE
Localize the output of run tests

### PR DESCRIPTION
**Pull request by dibarbet**

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+// <auto-generated>
+//     This code was generated by a tool.
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.CodeAnalysis.LanguageServer {
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.Designer.cs:11]: This file should not be generated. Designer files are source generated by Arcade as part of the build.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  <data name="Canceled" xml:space="preserve">
+    <value>Canceled!</value>
+  </data>
+  <data name="Discovering_tests" xml:space="preserve">
+    <value>Discovering tests...</value>
+  </data>
+  <data name="Failed" xml:space="preserve">
+    <value>Failed!</value>
+  </data>
+  <data name="Found_0_tests in_1" xml:space="preserve">
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx:135]:
```suggestion
  <data name="Found_0_tests_in_1" xml:space="preserve">
```

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj [hunk 3]
```diff
@@ -80,6 +80,21 @@
...
+      </Compile>
+    </ItemGroup>
+
+    <ItemGroup>
+      <EmbeddedResource Update="LanguageServerResources.resx">
+        <Generator>ResXFileCodeGenerator</Generator>
+        <LastGenOutput>LanguageServerResources.Designer.cs</LastGenOutput>
+      </EmbeddedResource>
+    </ItemGroup>
+
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj:97]:
```suggestion
    <ItemGroup>
      <EmbeddedResource Update="LanguageServerResources.resx" GenerateSource=true" />
    </ItemGroup>

```

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    <value>No test methods found in requested range</value>
+  </data>
+  <data name="Passed" xml:space="preserve">
+    <value>Passed!</value>
+  </data>
+  <data name="Running_tests" xml:space="preserve">
+    <value>Running tests...</value>
+  </data>
+  <data name="Stack_Trace" xml:space="preserve">
+    <value>Strack Trace</value>
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx:151]:
```suggestion
    <value>Stack Trace</value>
```

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  <data name="Starting_test_discovery" xml:space="preserve">
+    <value>Starting test discovery</value>
+  </data>
+  <data name="Starting_test_run" xml:space="preserve">
+    <value>Starting test run</value>
+  </data>
+  <data name="Summary" xml:space="preserve">
+    <value>Summary</value>
+  </data>
+  <data name="Summary_count" xml:space="preserve">
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServerResources.resx:162]:
```suggestion
  <data name="Failed_0_Passed_1_Skipped_2_Total_3_Duration_4" xml:space="preserve">
```

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/TestDiscoverer.cs [hunk 6]
```diff
@@ -76,7 +72,7 @@ internal partial class TestDiscoverer(ILoggerFactory loggerFactory)
 
         // Match what we found from vs test to what we found in the document to figure out exactly which tests to run.
         var matchedTests = await MatchDiscoveredTestsToTestsInRangeAsync(testCases, potentialTestMethods, testMethodFinder, document, cancellationToken);
-        progress.Report(partialResult with { Message = $"Found {matchedTests.Length} tests in {elapsed:g}" });
+        progress.Report(partialResult with { Message = string.Format(LanguageServerResources.Found_0_tests_in_1, matchedTests.Length, elapsed.ToString("g")) });
```
[the remark this answers is not in the document]
Comment (dibarbet) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/TestDiscoverer.cs]: Updated - now it outputs strings like
image

image

image

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/RunTestsHandler.cs [hunk 7]
```diff
@@ -66,6 +66,42 @@ public async Task<RunTestsPartialResult[]> HandleRequestAsync(RunTestsParams req
...
     }
 
+    /// <summary>
+    /// Format a timespan as a string similar to '5m 2s', omitting any value that is not present.
+    /// </summary>
+    internal static string GetShortTimespan(TimeSpan t)
+    {
+        var shortForm = "";
+        // Only output milliseconds if less than a second duration
+        if (t.Milliseconds > 0 && t.TotalSeconds < 1)
```
Comment (sharwell) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/RunTestsHandler.cs:76]: `0ms` is acceptable for time span 0.

```suggestion
        if (t.TotalSeconds < 1)
```

**Pull request merged**